### DISCUSSION TITLE
Does diesel support ordered-set aggregate functions in Postgres?

### DISCUSSION THREAD

**Discussion by davidatsurge** [diesel-rs/diesel#3562] [Q&A] [answered]
All in the title.

**Comment by weiznich** [comment, marked as answer]
That's not supported by the build-in dsl, but I guess it's something that could be implemented by an extension. 
Maybe we also add support for this to diesel itself when we add support for window functions.

**Reply by davidatsurge** [reply]
Thanks for the quick reply! 

I'll take a stab at implementing said extension and report back here.

**Comment by dnp1** [comment]
I'm implementing with AI help, since my personal projects are using many features from postgresql.

https://github.com/dnp1/diesel_pg_ext
